### DISCUSSION TITLE
Custom Patterns

### DISCUSSION THREAD

**Discussion by DawsonMacPhee** [winkjs/wink-nlp#49] [Help on coding]
When I create custom patterns and have the library tag text, is it possible for it to have one word tagged in two separate tags?

For example, I have a pattern for two nouns that are beside each other. If there was a situation that three nouns were beside each other, I'd like to be able to create two tags, for example:

"Alfred Joseph Hitchcock" would get tagged as "Alfred Joseph" and "Joseph Hitchcock"

However, I can't seem to get this to happen.

**Comment by sanjayaksaxena** [comment]
It is possible to have "one" word tagged under "two" separate tags as illustrated via the first example in documentation on [Custom Entities](https://winkjs.org/wink-nlp/custom-entities.html). This also highlights that It performs a greedy match and in case of multiple matches, the longest one is given preference.

If you can share your pattern with examples, we may be able to assist you further.

**Reply by DawsonMacPhee** [reply]
Here's an example of my patterns:

```
        {
            name: 'doubleNoun',
            patterns: ['[NOUN] [NOUN]', '[DET|ADJ] [NOUN]']
        },
        {
            name: 'doublePropn',
            patterns: ['[PROPN] [PROPN]', '[DET|ADJ] [PROPN]']
        },
        {
            name: 'doubleMix',
            patterns: ['[NOUN|PROPN] [NOUN|PROPN]']
        }
```
I can't seem to get them to overlap tags. With those patterns above, (for example)  "author Alfred Joseph" is getting tagged as "author Alfred", but the "Alfred Joseph" tag is getting missed.

**Reply by sanjayaksaxena** [reply]
Since it performs a greedy match, once "author Alfred" is detected the other pattern — "Alfred Joseph" can not be detected.
